src/tqec/computation/block_graph.py

```diff
@@ -452,6 +452,56 @@ def from_dae_file(filename: str | pathlib.Path, graph_name: str = "") -> BlockGr
 
         return read_block_graph_from_dae_file(filename, graph_name)
 
+    def relabel_cubes(
+        block_graph: BlockGraph, label_mapping: Mapping[Position3D | str, str]
```

Comment:
If that is a method, the first argument should be `self` which represents the instance this method is applied on (and is basically what you called `block_graph`):
```suggestion
        self, label_mapping: Mapping[Position3D | str, str]
```